### PR TITLE
add per worker env settings

### PR DESCRIPTION
**Pull request by lkevinzc**
- the env setting for new processes is important if we want to control some hardware access, e.g. "`CUDA_VISIBLE_DEVICES`"

### CODE REVIEW

File: mosec/coordinator.py [hunk 1]
```diff
@@ -188,3 +193,15 @@ def coordinate(self):
...
+_inheritance_format = """
+class MyWorker(Worker):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # initialize other resources
+        # ...
+
+    def forward(self, data: Any) -> Any:
+        # ...
+"""
```
Comment (kemingy) [inline mosec/coordinator.py:207]: Not necessary. We are not baby keepers.

File: mosec/coordinator.py [hunk 2]
```diff
@@ -54,10 +54,15 @@ def __init__(
             worker_id (int): identification number for worker processes at the same
                 stage.
         """
-        self.worker = worker()
+        try:
+            self.worker = worker(id=worker_id)
+        except TypeError as err:
```
Comment (kemingy) [inline mosec/coordinator.py:59]: I think the original exception is clear enough for the user to understand this.

File: examples/python_side_metrics.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 metric_registry = CollectorRegistry()
 multiprocess.MultiProcessCollector(metric_registry)
-counter = Counter("inference_result", "statistic of result", ("status", "pid"))
+counter = Counter("inference_result", "statistic of result", ("status", "workerId"))
```
Comment (kemingy) [inline examples/python_side_metrics.py:38]: snake case

File: mosec/server.py [hunk 4]
```diff
@@ -280,3 +280,25 @@ def run(self):
         except Exception:
             logger.error(traceback.format_exc().replace("\n", " "))
         self._halt()
+
+
+class _EnvContext(ContextDecorator):
```
Comment (kemingy) [inline mosec/server.py:285]: No need to add the `_` prefix.

File: mosec/server.py [hunk 5]
```diff
@@ -69,13 +72,35 @@ def _validate_arguments(
...
+                    if not (isinstance(k, str) and isinstance(v, str)):
+                        return False
+                return True
+
+            assert len(env) == num, "len(env) must equal to num"
+            valid = True
+            if not isinstance(env, List):
+                valid = False
+            elif not all(
+                [lambda x: isinstance(x, Dict) and validate_str_dict(x) for x in env]
```
Comment (kemingy) [inline mosec/server.py:98]: Check your Pylance warning here.

Comment (lkevinzc) [reply mosec/server.py:98]: can give a screenshot on that? I don't see warning here

Comment (kemingy) [reply mosec/server.py:98]: You created a lambda function but you never call it.

Comment (kemingy) [inline mosec/server.py:98]:
```suggestion
                [isinstance(x, Dict) and validate_str_dict(x) for x in env]
```

**Review by kemingy — APPROVED** [review]

**Pull request merged**